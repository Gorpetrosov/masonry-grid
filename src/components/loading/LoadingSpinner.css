.spinner-container {
    opacity: 0;
    transition: opacity 0.3s ease;
    pointer-events: none;
}

.spinner-container.spinner-flex {
    display: flex;
    justify-content: center;
    align-items: center;
    opacity: 1;
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: rgba(255, 255, 255, 0.8);
    z-index: 111111;
}

.spinner-icon {
    animation: spin 1s linear infinite;
}

@keyframes spin {
    from { transform: rotate(0deg); }
    to { transform: rotate(360deg); }
}