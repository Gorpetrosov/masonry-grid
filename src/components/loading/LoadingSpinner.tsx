import { ImSpinner8 } from 'react-icons/im';
import "./LoadingSpinner.css"

interface LoadingSpinnerProps {
    isLoading: boolean;
    size?: number;
    color?: string;
}

export const LoadingSpinner = ({
                                   size = 40,
                                   color = '#ed2b2b',
                                   isLoading
                               }: LoadingSpinnerProps) => {
    return (
        <div className={`spinner-container ${isLoading ? 'spinner-flex' : ''}`}>
                <ImSpinner8
                    className="spinner-icon"
                    style={{ fontSize: size, color }}
                />
        </div>
    );
};