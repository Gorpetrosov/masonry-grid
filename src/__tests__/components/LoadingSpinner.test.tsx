import { render, screen } from '@testing-library/react';
import { describe, test, expect, vi } from 'vitest';
import {LoadingSpinner} from "../../components/loading/LoadingSpinner.tsx";

vi.mock('react-icons/im', () => ({
    ImSpinner8: (props: any) => <div data-testid="spinner-icon" {...props} />,
}));

describe('LoadingSpinner Component', () => {
    test('renders spinner when isLoading is true', () => {
        render(<LoadingSpinner isLoading={true} />);

        const container = screen.getByTestId('spinner-icon').parentElement;
        expect(container).toHaveClass('spinner-container', 'spinner-flex');
    });

    test('does not render spinner when isLoading is false', () => {
        render(<LoadingSpinner isLoading={false} />);

        const container = screen.getByTestId('spinner-icon').parentElement;
        expect(container).toHaveClass('spinner-container');
        expect(container).not.toHaveClass('spinner-flex');
    });

    test('applies custom size and color', () => {
        const testSize = 50;
        const testColor = '#4287f5';

        render(<LoadingSpinner isLoading={true} size={testSize} color={testColor} />);

        const icon = screen.getByTestId('spinner-icon');
        expect(icon).toHaveStyle({
            fontSize: `${testSize}px`,
            color: testColor
        });
    });

    test('uses default props when not provided', () => {
        render(<LoadingSpinner isLoading={true} />);

        const icon = screen.getByTestId('spinner-icon');
        expect(icon).toHaveStyle({
            fontSize: '40px',
            color: '#ed2b2b'
        });
    });

    test('matches snapshot when visible', () => {
        const { container } = render(<LoadingSpinner isLoading={true} />);
        expect(container).toMatchSnapshot();
    });

    test('matches snapshot when hidden', () => {
        const { container } = render(<LoadingSpinner isLoading={false} />);
        expect(container).toMatchSnapshot();
    });
});