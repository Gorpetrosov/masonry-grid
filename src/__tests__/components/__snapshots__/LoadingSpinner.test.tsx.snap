// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingSpinner Component > matches snapshot when hidden 1`] = `
<div>
  <div
    class="spinner-container "
  >
    <div
      class="spinner-icon"
      data-testid="spinner-icon"
      style="font-size: 40px; color: rgb(237, 43, 43);"
    />
  </div>
</div>
`;

exports[`LoadingSpinner Component > matches snapshot when visible 1`] = `
<div>
  <div
    class="spinner-container spinner-flex"
  >
    <div
      class="spinner-icon"
      data-testid="spinner-icon"
      style="font-size: 40px; color: rgb(237, 43, 43);"
    />
  </div>
</div>
`;
